app/controllers/application_controller.rb

```diff
@@ -29,4 +30,25 @@ def claim_session_timed_out?
   def claim_timeout_in_minutes
     self.class::CLAIM_TIMEOUT_LENGTH_IN_MINUTES
   end
+
+  def end_expired_admin_sessions
```

Comment:
I'm a bit uneasy about moving the logic for claim and admin session expiry into the `ApplicationController` superclass. It increases the method surface area of its subclasses, introducing methods that do not make sense for that subclass (for example, should `BasePublicController` have access to a method called `admin_session_timed_out?`?)

I don't know whether this is a worry for you, and if not then no problem. But one alternative would be to introduce concerns to contain the specific expiry methods (I think is the Rails thing, right?):

- `AdminSessionTimeout`: included in `BaseAdminController` and `SessionsController`;
- `ClaimSessionTimeout`: included in `BasePublicController` and `SessionsController`.

Reply:
Fair point, having them separated out like that will reduce the chance of admin session-related methods being used in the public-facing part of the site and vice versa 👍🏽